integration/e2e/metrics_promql.go

```diff
@@ -0,0 +1,283 @@
+						ts:     getTimestamp(ts, m.GetTimestampMs()),
+						val:    float64(b.GetCumulativeCount()),
+					})
+				}
+
+				result = append(result, sample{
+					labels: getLabels(mf.GetName()+"_count", m.GetLabel()),
+					ts:     getTimestamp(ts, m.GetTimestampMs()),
+					val:    float64(m.GetHistogram().GetSampleCount()),
+				})
+
+				result = append(result, sample{
+					labels: getLabels(mf.GetName()+"_sum", m.GetLabel()),
+					ts:     getTimestamp(ts, m.GetTimestampMs()),
+					val:    m.GetHistogram().GetSampleSum(),
+				})
+			}
+		}
+	}
+
+	return result
+}
+
+func getTimestamp(ts int64, mts int64) int64 {
+	if mts != 0 {
+		return mts
+	}
+	return ts
+}
+
+func getLabels(name string, pairs []*dto.LabelPair) map[string]string {
+	return getLabelsExt(name, pairs, false, 0)
+}
+
+func getLabelsExt(name string, pairs []*dto.LabelPair, bucket bool, upperBound float64) map[string]string {
+	result := map[string]string{}
+	result[labels.MetricName] = name
+	if bucket {
+		result[labels.BucketLabel] = fmt.Sprintf("%g", upperBound)
+	}
+
+	for _, p := range pairs {
+		result[p.GetName()] = p.GetValue()
+	}
+	return result
+}
+
+// implements Queryable and Querier
+type samplesQueryable []sample
+
+func (sq *samplesQueryable) Querier(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
+	return sq, nil
+}
+
+func (sq samplesQueryable) SelectSorted(params *storage.SelectParams, matcher ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
+	var series []storage.Series
+
+	for _, s := range sq {
+		if s.matches(matcher...) {
+			series = append(series, s)
+		}
+	}
+
+	return &seriesSet{series: series}, nil, nil
+}
+
+func (sq samplesQueryable) Select(params *storage.SelectParams, matcher ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
+	return sq.SelectSorted(params, matcher...)
+}
+
+func (sq *samplesQueryable) LabelValues(name string) ([]string, storage.Warnings, error) {
+	panic("implement me")
+}
+
+func (sq *samplesQueryable) LabelNames() ([]string, storage.Warnings, error) {
+	panic("implement me")
+}
+
+func (sq *samplesQueryable) Close() error {
+	return nil
+}
+
+// seriesSet implements storage.SeriesSet.
+type seriesSet struct {
+	cur    int
+	series []storage.Series
+}
+
+func (c *seriesSet) Next() bool {
+	c.cur++
+	return c.cur-1 < len(c.series)
+}
+
+func (c *seriesSet) At() storage.Series {
+	return c.series[c.cur-1]
+}
+
+func (c *seriesSet) Err() error {
+	return nil
+}
+
+// sample implements storage.Series
+func (c sample) Labels() labels.Labels {
+	b := labels.NewBuilder(nil)
+	for k, v := range c.labels {
+		b.Set(k, v)
+	}
+	return b.Labels()
+}
+
+func (c sample) Iterator() storage.SeriesIterator {
+	return newSampleIterator(c.ts, c.val)
+}
+
+func (c sample) matches(matcher ...*labels.Matcher) bool {
+	for _, m := range matcher {
+		if !m.Matches(c.labels[m.Name]) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// concreteSeriesIterator implements storage.SeriesIterator.
```

Comment:
wrong comment (should fixed and either full sentence or removed and you can add validating that)